services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipInvitationManagerImpl.java

```diff
@@ -193,46 +196,50 @@ public Count getOpenInvitationCountForUser(String principalId) {
 	}
 
 	@Override
-	public InviteeVerificationSignedToken verifyInvitee(Long userId, String membershipInvitationId) {
+	public InviteeVerificationSignedToken verifyInvitee(Long userId, String membershipInvitationId, MembershipInvtnSignedToken token) {
 		ValidateArgument.required(userId, "userId");
 		ValidateArgument.required(membershipInvitationId, "membershipInvitationId");
-		// Get list of email addresses associated with userId
-		List<PrincipalAlias> aliases = principalAliasDAO.listPrincipalAliases(userId);
-		List<String> emails = new ArrayList<>();
-		for (PrincipalAlias alias : aliases) {
-			if (alias.getType() == AliasType.USER_EMAIL) {
-				emails.add(alias.getAlias());
-			}
+		SignedTokenUtil.validateToken(token);
+		if (!membershipInvitationId.equals(token.getMembershipInvitationId())) {
+			throw new IllegalArgumentException("ID in URI and ID in signed token don't match");
+		}
+		if (token.getExpiresOn() != null && token.getExpiresOn().before(new Date())) {
+			throw new IllegalArgumentException("MembershipInvtnSignedToken is expired");
 		}
 		// Get inviteeEmail from membershipInvitation
 		String inviteeEmail = membershipInvtnSubmissionDAO.getInviteeEmail(membershipInvitationId);
-		// If inviteeEmail is in the emails list, construct InviteeVerificationSignedToken and return it
-		if (emails.contains(inviteeEmail)) {
-			InviteeVerificationSignedToken token = new InviteeVerificationSignedToken();
-			token.setInviteeId(userId.toString());
-			token.setMembershipInvitationId(membershipInvitationId);
-			SignedTokenUtil.signToken(token);
-			return token;
+		// Find out if the given user has an alias that matches the inviteeEmail
+		PrincipalAlias principalAlias = principalAliasDAO.findPrincipalWithAlias(inviteeEmail);
```

Comment:
I would like to see a test for the case in which the email associated with the user is "xxx.com" and the email entered in the invitation is "xxxx.COM", i.e. the normalized alias matches but the alias entered by the users does not.